src/cata_utility.cpp

```diff
@@ -42,6 +42,40 @@ bool lcmatch( const std::string &str, const std::string &qry )
     return haystack.find( needle ) != std::string::npos;
 }
 
+bool match_include_exclude( const std::string &text, std::string filter )
+{
+    unsigned long iPos;
```

Comment:
Changing this from `unsigned long` to `size_t` resolves hang in VS and MSYS compiles just fine.